src/frontend/src/optimizer/plan_node/generic/source.rs

```diff
@@ -28,20 +29,37 @@ use crate::optimizer::optimizer_context::OptimizerContextRef;
 use crate::optimizer::property::FunctionalDependencySet;
 use crate::{TableCatalog, WithOptions};
 
+/// In which scnario the source node is created
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[expect(clippy::enum_variant_names)]
+pub enum SourceNodeKind {
```

Comment:
can you add another field to distinguish cdc source?

Reply:
I think cdc source can just use CreateSourceWithStreamjob